server/honeybadger/plugins.py

```diff
@@ -26,18 +28,51 @@ def get_coords_from_google(aps):
         data['lng'] = jsondata['location']['lng']
     return data
 
-def get_coords_from_uniapple(ip):
-    logger.info('Geolocating via Uniapple API.')
-    url = 'http://uniapple.net/geoip/?ip={}'.format(ip)
-    content = urllib2.urlopen(url).read()
-    logger.info('Uniapple API response:\n{}'.format(content))
+def get_external_ip(ip):
+    logger.info('Resolving external IP via Ipify API.')
+    url = "https://api.ipify.org"   # IPs are often reported as internal when testing,
+                                    #   and can't be geolocated by ipstack or ipinfo
+                                    #   as a result. This API call remedies that.
+    content = urlopen(url).read()
+    return content.decode()
```

Comment:
Why does this need to be decoded? Seems to be that Flask should be handling the the encoding stuff internally. Not saying it isn't needed, just curious as to why you need to do it here.

Reply:
Similar to the other comment about needing to encode, I recall type errors when not decoding. I believe urllib changed the encoding between python 2 and python 3, so the function would crash when not decoding.

In my Python shell, running that bit of code and then running `type(content)` gives `<class 'bytes'>`.